libportal/portal-helpers.h

```diff
@@ -35,6 +35,27 @@ XDP_PUBLIC
 G_DECLARE_FINAL_TYPE (XdpPortal, xdp_portal, XDP, PORTAL, GObject)
 
 XDP_PUBLIC
-XdpPortal *xdp_portal_new                    (void);
+XdpPortal *xdp_portal_new                       (void);
+
+XDP_PUBLIC
+gboolean   xdp_portal_pid_running_under_flatpak (pid_t    pid,
```

Comment:
Who is the consumer of these pid APIs?

I'm just not sure any project, other than the portals themselves, care about this in practice? And I just think this library is for client applications rather than portal services.

Reply:
I was thinking xdg-desktop-portal but it looks like there is at least one other potential user because Tracker has also copied [this code](https://github.com/GNOME/tracker/blob/5e612b2e55d569b8e5b228a124cf59a36d200e3d/src/portal/tracker-portal-utils.c#L156) from x-d-p. But tracker reads the key file so we'd need to add that here if we want to support that use case.

Maybe there are other system components that need to know whether to treat a process as trusted. You think it's better to get rid of the pid APIs here?

Reply:
Yes, I think we should only add public APIs that have clear users. Since it also can greatly simplify the code removing it, I think its best to remove it.

Reply:
The original goal for libportal is to be used by applications, not other services. This PID-based API did seem a bit out of place to me too.

Reply:
Alright, I removed the pid APIs.